### ISSUE TITLE
Sidenav does not set 'active' on current route, does not support routerLinkActive

### ISSUE THREAD

**Issue by wnipper** [sherweb/ng2-materialize#190]
When you select a link in MaterializeCss's sidenav, the link is set to active.   The ng2-materialize sidenav doesn't automatically set the class for a newly selected link and doesn't appear to expose any way for the developer to do so.  The standard way to do that in angular is to set the class for the item with `routerLinkActive="['className']"`. 

Proposal: replace `<li [class.active]="active">` with `<li [routerLInkActive]="['active']">` in [sidenav-link.component.ts](https://github.com/sherweb/ng2-materialize/blob/master/src/app/sidenav/sidenav-link/sidenav-link.component.html) and add the import for `angular/router` to the component.ts.

**Comment by wnipper** [comment]
In case anyone else experiences this issue, the workaround I used was to fill the list item with the anchor and set the class on the anchor.  This doesn't hit all use cases, but it does work as expected.  Example:
`<mz-sidenav-link>
      <a routerLink="dashboard" routerLinkActive="active">Dashboard</a>
    </mz-sidenav-link>
  <mz-sidenav-link>
      <a routerLink="users" routerLinkActive="active">Users</a>
  </mz-sidenav-link>
    <mz-sidenav-link>
      <a routerLink="groups" routerLinkActive="active">Groups</a>
    </mz-sidenav-link>
    <mz-sidenav-link>
      <a routerLink="roles" routerLinkActive="active">Roles</a>
    </mz-sidenav-link>`

**Comment by jfcere** [comment]
Hi @wnipper,

Thanks for the workaround. We're gonna have to take a look at it before to take action.

Just wondering what you meant by "it doesn't his all use cases" ?
This could be helpfull for the fix investigation.

**Comment by IAMtheIAM** [comment]
I agree with @wnipper It should hook into the Angular routerLinkActive directive in order to set the class. Setting it manually doesn't really make too much sense. `<li routerLinkActive="active">` this is a good solution.

**Comment by IAMtheIAM** [comment]
I was able to modify the sidenav component to get it working properly with no workaround, using angular's native implementation. I made a PR, please review https://github.com/sherweb/ng2-materialize/pull/226

**Comment by jfcere** [comment]
I think the better fix would be the **tomscholz** proposition on the Materialize issue you created. Allowing `active` class to be on the `a` tag would solve the problem without having to change anything.

Adding `routerLink` property binds the component `@angular/router` and does not help people that would use alternate router (god knows why they would)

**Comment by IAMtheIAM** [comment]
OK, no problem. I'll update here when they respond to that issue.

**Comment by IAMtheIAM** [comment]
It looks like we'll be good to go, with the change suggested at Materialize. It will allow `.active` to apply to any element instead of `li` or `a`. Then this:

```html
<mz-sidenav-link>
  <a routerLink="dashboard" routerLinkActive="active">Dashboard</a>
</mz-sidenav-link>
<mz-sidenav-link>
  <a routerLink="users" routerLinkActive="active">Users</a>
</mz-sidenav-link>
```
will work as expected.
